### PR TITLE
Compact Graph Structure 

### PR DESCRIPTION
**Pull request by KhanWhale**
Revamp graph structure:
Rather than having A + PR_B -> C and B + PR_A -> C reaction nodes, compress into just one A + B -> C node and abstract away PR/Reactant identification. 
Still need to update concerteds and rnmc (due to concerted dependence).

### CODE REVIEW

File: src/mrnet/network/reaction_network.py [hunk 1]
```diff
@@ -806,7 +928,7 @@ def parse_path(self, path):
         for step in path:
             if isinstance(step, int):
                 nodes.append(step)
-            elif "PR_" in step:
+            elif "PR_" in step:  # flag
```
Comment (samblau) [inline src/mrnet/network/reaction_network.py:931]: So first off, I don't totally remember when parse_path was introduced, and it seems weird to me that we have both parse_path and characterize_path. But also, I believe you said that PR solving was all working, and yet parse_path is still checking for "PR_"? Does that mean that parse_path isn't necessary to PR solving?

File: src/mrnet/network/reaction_network.py [hunk 2]
```diff
@@ -1010,6 +1144,35 @@ def find_path_cost(
                             self.graph,
                             old_solved_PRs,
                         )
+                        if (
```
Comment (samblau) [inline src/mrnet/network/reaction_network.py:1147]: Again, sad times. At least wrap this in a function or something?

File: src/mrnet/network/reaction_network.py [hunk 3]
```diff
@@ -1163,7 +1330,7 @@ def remove_node(self, node_ind: List[int]):
             for node in nodes:
                 if self.graph.nodes[node]["bipartite"] == 1:
                     reactants = node.split(",")[0].split("+")
-                    reactants = [reac.replace("PR_", "") for reac in reactants]
+                    reactants = [reac.replace("PR_", "") for reac in reactants]  # flag
```
Comment (samblau) [inline src/mrnet/network/reaction_network.py:1333]: Can this line just be removed? It seems like the previous line should work as intended given "A+B,C" or "A,B"

File: tests/network/test_reaction_network.py [hunk 4]
```diff
@@ -357,57 +333,90 @@ def test_solve_prerequisites(self):
...
-                    self.assertAlmostEqual(
-                        PRs_calc[node][start].cost, PR_paths[node][start].cost, places=2
-                    )
-                if PRs_calc[node][start].pure_cost != PR_paths[node][start].pure_cost:
-                    self.assertAlmostEqual(
-                        PRs_calc[node][start].pure_cost,
-                        PR_paths[node][start].pure_cost,
-                        places=2,
                     )
+                    for i in range(
```
Comment (samblau) [inline tests/network/test_reaction_network.py:364]: This chunk could use some comments. I mostly get what's happening here, but a year in the future it's possible I'd find it totally incomprehensible

**Review by samblau — CHANGES_REQUESTED** [review]

**Comment by KhanWhale** [comment]
RNMC and concerteds updated as well.

**Review by hpatel1567 — APPROVED** [review]

**Review by samblau — APPROVED** [review]

**Comment by samblau** [comment]
Once Linting passes, this can be merged! Wouldn't hurt to pull in Main one more time once you're all done. Thanks again for your great work!

**Pull request merged**